### PR TITLE
adding the ray serve article

### PR DESCRIPTION
**Pull request by TuanaCelik**
I went ahead and created a pr as it's nice to see the preview. We can change anything...

### CODE REVIEW

File: content/blog/run-haystack-pipelines-with-ray-serve/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
+---
+layout: blog-post
+title: Run Haystack pipelines in production with Ray Serve
```
Comment (bilgeyucel) [inline content/blog/run-haystack-pipelines-with-ray-serve/index.md:3]: How about this title 👇 
```suggestion
title: Run Haystack Pipelines in Production with Ray Serve
```

File: content/blog/run-haystack-pipelines-with-ray-serve/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+haystack_deployment = HaystackQA.bind()
+
+```
+
+A couple of things to note about the code above:
+
+-   We added a `__init__` method where we initialize the pipeline once and for the whole lifetime of the deployment.
+-   Every class decorated with `@serve.deployment` gets a `bind` method that tells Ray Serve to attach our `HaystackQA` class to a deployment object that we name `haystack_deployment`. In a moment, when we will start the server process, Ray Serve will pick up the value of this variable.
+
+The content of the Python file `[capitals.py](<http://capitals.py>)` is now all we need to “serve” our deployment. Ray Serve can be run from Python directly, but for this example we’ll show how to start the process from the command line, which is closer to what we would do in a production use case. From a shell where the Python environment containing `ray` is active, you can just run:
```
Comment (bilgeyucel) [inline content/blog/run-haystack-pipelines-with-ray-serve/index.md:134]:
```suggestion
The content of the Python file `capitals.py` is now all we need to “serve” our deployment. Ray Serve can be run from Python directly, but for this example we’ll show how to start the process from the command line, which is closer to what we would do in a production use case. From a shell where the Python environment containing `ray` is active, you can just run:
```

File: content/blog/run-haystack-pipelines-with-ray-serve/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+```python
+@serve.deployment
+class MinimalIngress:
+  async def __call__(self, request: Request) -> str:
+      name = await request.json()["name"]
+      return f"Hello {name}"
+
+```
+
+The presence of an `async def __call__` method is the only contract we need to fulfil in order to spawn a Ray Serve deployment. To create a deployment that will run a Haystack pipeline, let’s create a file called `[capitals.py](<http://capitals.py>)` and add the following code:
```
Comment (bilgeyucel) [inline content/blog/run-haystack-pipelines-with-ray-serve/index.md:98]:
```suggestion
The presence of an `async def __call__` method is the only contract we need to fulfill in order to spawn a Ray Serve deployment. To create a deployment that will run a Haystack pipeline, let’s create a file called `capitals.py` and add the following code:
```

File: content/blog/run-haystack-pipelines-with-ray-serve/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+The `autoscaling_config` parameters are self-explanatory and the [docs](https://docs.ray.io/en/latest/serve/scaling-and-resource-allocation.html#autoscaling-config-parameters) are very helpful in detailing the logic behind that. In this case we tell Ray Serve that we want to scale up our deployment to up to 5 deployment replicas and scale down to at least 1. We also want to start with 2 replicas and downscale if there’s no activity for 30 seconds. Once again you can observe how this works behind the scenes by starting the server and looking at the dashboard on [`http://127.0.0.1:8265/#/actors`](http://127.0.0.1:8265/#/actors).
+
+## What’s next
+
+The deployment strategy we introduced in this article is ok for many use cases ranging from trying out Haystack to demo applications and small POCs, but close readers will note the limit of scaling up and down a whole Haystack pipeline: certain nodes of a pipeline might benefit more than others of horizontal scaling and redundancy, can we do that with Ray Serve? The answer is yes, and we’ll see in an upcoming article how we can split a pipeline into multiple deployments that can be then managed by Ray Serve with different strategies
```
Comment (bilgeyucel) [inline content/blog/run-haystack-pipelines-with-ray-serve/index.md:185]:
```suggestion
The deployment strategy we introduced in this article is ok for many use cases ranging from trying out Haystack to demo applications and small POCs, but close readers will note the limit of scaling up and down a whole Haystack pipeline: certain nodes of a pipeline might benefit more than others of horizontal scaling and redundancy, can we do that with Ray Serve? The answer is yes, and we’ll see in an upcoming article how we can split a pipeline into multiple deployments that can be then managed by Ray Serve with different strategies.
```

File: content/blog/run-haystack-pipelines-with-ray-serve/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+```docker
+docker run -p 9200:9200 -p 9300:9300 -d deepset/elasticsearch-countries-and-capitals:latest
+
+```
+
+The Elasticsearch instance will accept connections on `[localhost](<http://localhost>)` through the port `9200`.
+
+### Run the Haystack pipeline
+
+Haystack pipelines can be defined through Python code or a yaml definition that is eventually and transparently converted into Python code. We’ll use the latter format, so open an editor and save the following definition in a file called `pipeline.yml`
```
Comment (bilgeyucel) [inline content/blog/run-haystack-pipelines-with-ray-serve/index.md:33]:
```suggestion
Haystack pipelines can be defined through Python code or a [yaml definition](https://docs.haystack.deepset.ai/docs/pipelines#yaml-file-definitions) that is eventually and transparently converted into Python code. We’ll use the latter format, so open an editor and save the following definition in a file called `pipeline.yml`
```

**Review by masci — APPROVED** [review]
Thanks for the review! 🙏

**Pull request merged**